gengokit/template/NAME-service/generated/endpoints.gotemplate

```diff
@@ -63,14 +63,24 @@ type Endpoints struct {
 // WrapAll wraps each Endpoint field of struct Endpoints with a
 // go-kit/kit/endpoint.Middleware.
 // Use this for applying a set of middlewares to every endpoint in the service.
-// The middlewares will be applied in the order passed, with the first
-// middleware being the outermost middleware.
-func (e *Endpoints) WrapAll(middlewares ...endpoint.Middleware) {
-	if len(middlewares) == 0 {
-		return
+// Optionally, endpoints can be passed in by name to be excluded from being wrapped.
+// WrapAll(middleware, "Status", "Ping")
+func (e *Endpoints) WrapAll(middleware endpoint.Middleware, excluded ...string) {
+	included := map[string]struct{}{
+		{{- range $i := .Service.Methods}}
+			"{{$i.Name}}": struct{}{},
+		{{- end}}
+	}
+
+	for _, ex := range excluded {
+		delete(included, ex)
```

Comment:
@ian the user is passing in Endpoints by name. There was no good way to get the struct field name, it is lost in the function call. I could explain in person, though there are a lot of particulars. Zaq and I ran through the options this morning.

We are using panic now.